### ISSUE TITLE
Pipfile: the replacement for requirements.txt

### ISSUE THREAD

**Issue by Kaushl2208** [fossology/atarashi#66]
We can migrate our current requirement.txt file to pipfile due to several reasons like:

- TOML syntax for declaring all types of Python dependencies.
- One Pipfile (as opposed to multiple requirements.txt files).
- A Pipfile is inherently ordered.

or, Refer: [Why?](https://github.com/pypa/pipfile#the-concept)

**Comment by GMishx** [comment]
That would be a good change. But will it effect the existing [pyproject.toml](https://github.com/fossology/atarashi/blob/4064aaff6e76da0d94c0e9abf709758e4ee107d9/pyproject.toml) (compatibility with [PEP 518](https://www.python.org/dev/peps/pep-0518/))?

**Comment by Kaushl2208** [comment]
Yes, It will replace pyproject.toml and it might affect setup.py. Not sure, about it rn.

Can I submit a PR for it?

**Comment by GMishx** [comment]
Yes, sure @Kaushl2208. Just make sure the current build wheel build should not break because of this change.

**Comment by lohithmunakala** [comment]
Is this still open @GMishx ? I wanna work on this.

**Comment by GMishx** [comment]
Yes @lohithmunakala, from my end the issue is still open. @Kaushl2208 do let us know if you are working on the issue.

It will be nice to assign the issue to the person currently working on it.

**Comment by lohithmunakala** [comment]
If @Kaushl2208 isn't working on it rn, can you help me get started with the issue @GMishx ?

**Comment by Kaushl2208** [comment]
Hey @GMishx, We can assign this issue to @lohithmunakala. I don't have the rights to do so.

**Comment by hastagAB** [comment]
> Hey @GMishx, We can assign this issue to @lohithmunakala. I don't have the rights to do so.

Meanwhile @lohithmunakala You can start working on this issue. It will be a great help.

**Comment by GMishx** [comment]
I have assigned the ticket to you @lohithmunakala .

You can start looking into pipenv documentation: https://pipenv.pypa.io/en/stable/
Expectation from this change is:
1. Removal of dependencies from requirements.txt and setup.py
1. Should not break [PEP 517](https://www.python.org/dev/peps/pep-0517/) changes (try to build Dockerfile to test)
1. Clear separation of additional dependencies like `csv` and `json` files generated by running `python3 setup.py build`

**Comment by ShivamGupta-16** [comment]
Can I work on this issue if it is open ?

**Comment by aviralgarg05** [comment]
I would like to solve the issue, please assign me

**Comment by Kasturijadhav14** [comment]
heyy!! I am kasturi jadhav, a cs student from India . applying for GSOC 2026. I will like to work on this issue as a part of my contribution to FOSSology.

**Comment by Kasturijadhav14** [comment]
Hey, I went through this issue and added a Pipfile and 
Pipfile.lock to the project in PR #119.

All the dependencies from pyproject.toml are included 
with the same version ranges. Added a [dev-packages] 
section as well for future dev dependencies.

You can use it like this:
pip install pipenv
pipenv install
pipenv run python atarashi/build_deps.py

Screenshot of the Pipfile attached below.

<img width="1920" height="1020" alt="Image" src="https://github.com/user-attachments/assets/685cdb2b-fce2-443c-90a5-32046598a60e" />

**Comment by Abh-igyan** [comment]
Hi @Kaushl2208 @GMishx I am new here and earlier contrinbuted to an[ issue](https://github.com/fossology/atarashi/pull/120) for Atarashi and want to contribute to this issue. Can I be assigned this issue? No PRs seem linked to it!
